ShoppingList/ShoppingList/MainWindowViewModel.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Windows.Input;
+
+namespace ShoppingList
+{
+    public class MainWindowViewModel : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void SetProperty<T>(ref T field, T value, [CallerMemberName]string propertyName = null)
+        {
+            if (!EqualityComparer<T>.Default.Equals(field, value))
+            {
+                field = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            }
+        }
+
+        public ObservableCollection<ShoppingItem> ShoppingItemList { get; } = new ObservableCollection<ShoppingItem>();
+
+        public Command AddItemCommand { get; }
+        public Command DeleteItemCommand { get; }
+
+        private bool CanExecute { get; set; }
+
+        private ShoppingItem _SelectedShoppingItem;
```

Comment:
```suggestion
        private ShoppingItem _SelectedShoppingItem = new ShoppingItem("Item", true, 1, 1, UnitType.Each);
```

Just a throw-away item. Safer than using null! or leaving it uninitialized, in case you don't initialize it and the list in the constructor.